eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/EnforceOverrideAnnotationCheck.java

```diff
@@ -0,0 +1,141 @@
+     */
+    private String className;
+
+    /**
+     * {@link Override Override} annotation name.
+     */
+    private static final String OVERRIDE = "Override";
+
+    /**
+     * Fully-qualified {@link Override Override} annotation name.
+     */
+    private static final String FQ_OVERRIDE = "java.lang." + OVERRIDE;
+
+    private static final String ANNOTATION_MISSING_OVERRIDE = "Must include @java.lang.Override or @Override annotation";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[]{TokenTypes.PACKAGE_DEF, TokenTypes.CLASS_DEF, TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                packageName = extractQualifiedName(token.getFirstChild().getNextSibling());
+                break;
+            case TokenTypes.CLASS_DEF:
+                className = token.findFirstToken(TokenTypes.IDENT).getText();
+                break;
+            case TokenTypes.METHOD_DEF:
+                String methodName = token.findFirstToken(TokenTypes.IDENT).getText();
+
+                if (isMethodOverridden(methodName, getMethodParameters(token)) &&
+                    !AnnotationUtil.containsAnnotation(token, OVERRIDE) &&
+                    !AnnotationUtil.containsAnnotation(token, FQ_OVERRIDE)) {
+                    log(token, ANNOTATION_MISSING_OVERRIDE);
+                }
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Get name of class (with qualified package if specified) in {@code ast}.
+     *
+     * @param ast ast to extract class name from
+     * @return qualified name
+     */
+    private static String extractQualifiedName(DetailAST ast) {
+        return FullIdent.createFullIdent(ast).getText();
+    }
+
+    private Class<?>[] getMethodParameters(DetailAST token) {
+        DetailAST parameters = token.findFirstToken(TokenTypes.PARAMETERS);
+        DetailAST child = parameters.findFirstToken(TokenTypes.PARAMETER_DEF);
+        List<Class<?>> classes = new ArrayList<>();
+        while (child != null) {
+            if (child.getType() == TokenTypes.PARAMETER_DEF) {
+                classes.add(child.findFirstToken(TokenTypes.TYPE).getFirstChild().getClass());
+            }
+            child = child.getNextSibling();
+        }
+        Class<?>[] params = new Class<?>[classes.size()];
+        return classes.toArray(params);
+    }
+
+    private boolean isMethodOverridden(String name, Class<?>... parameterTypes) {
+
+        Class<?> klass;
+        try {
+            klass = Class.forName(packageName + PACKAGE_SEPARATOR + className);
```

Comment:
Hi @alzimmermsft, I am having some issues over here, it always ends up in `catch` block
I am using the following commands for now:
```
mvn install -f eng/code-quality-reports/pom.xml
mvn install -f sdk/cosmos/azure-cosmos/pom.xml -Dgpg.skip -Drevapi.skip -DskipTests
```
An example error:
```
java.lang.ClassNotFoundException: com.azure.cosmos.CosmosTrigger
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:315)
	at com.azure.tools.checkstyle.checks.EnforceOverrideAnnotationCheck.isMethodOverridden(EnforceOverrideAnnotationCheck.java:115)
```
Please help me here.

Reply:
For environment details:
```
$ java -version
openjdk version "11.0.8" 2020-07-14 LTS
OpenJDK Runtime Environment Zulu11.41+24-SA (build 11.0.8+10-LTS)
OpenJDK 64-Bit Server VM Zulu11.41+24-SA (build 11.0.8+10-LTS, mixed mode)

$ mvn -version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/nitikaga/apache-maven-3.6.3
Java version: 11.0.8, vendor: Azul Systems, Inc., runtime: /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home
Default locale: en_IN, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.4", arch: "x86_64", family: "mac"
```

Reply:
The first issue that I see is that this codesnippet will get `DetailAst.class`.

```java
if (child.getType() == TokenTypes.PARAMETER_DEF) {
    classes.add(child.findFirstToken(TokenTypes.TYPE).getFirstChild().getClass());
}
```

I'd recommend creating unit tests for this Checkstyle so you're able to debug a few common scenarios more easily (and have a faster dev-feedback loop).

https://github.com/Azure/azure-sdk-for-java/blob/master/eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/JavadocThrowsChecksTest.java